File: leetcode/1768_merge_strings_alternaly/solution.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int n = word1.size();
        int m = word2.size();

        string result;
        result.reserve(n + m);

        int i = 0;
        while (i < n && i < m) {
            result.push_back(word1[i]);
            result.push_back(word2[i]);
            i++;
        }

        if (i < n) {
            result.append(word1.substr(i));
        } else if (i < m) {
            result.append(word2.substr(i));
        }

        return result;
    }
};

File: leetcode/1768_merge_strings_alternaly/solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        n, m = len(word1), len(word2)
        i, j = 0, 0
        parts = []

        while i < n and j < m:
            parts.append(word1[i])
            parts.append(word2[j])
            i += 1
            j += 1

        if i < n:
            parts.append(word1[i:])
        elif j < m:
            parts.append(word2[j:])

        return ''.join(parts)
